sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcFileFormat.scala

```diff
@@ -194,7 +194,7 @@ class OrcFileFormat extends FileFormat with DataSourceRegister with Serializable
   }
 
   override def supportDataType(dataType: DataType): Boolean = dataType match {
-    case _: AnsiIntervalType => false
+    case _: AnsiIntervalType | _: TimestampNTZType => false
```

Comment:
QQ: Why do we disallow AnsiIntervalType and TimestampNTZType in Hive ORC file format?

Reply:
I guess because Hive MetaStore/SerDe doesn't support Hive's interval types. For example, interval types are not handled there: https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/io/orc/OrcStruct.java#L510

Reply:
Because we store TimestampNTZ as timestamp type in ORC, I will revert this change.